File: swift/easy/203-remove-linked-list-elements.swift
func removeElements(_ head: ListNode?, _ val: Int) -> ListNode? {
    var dummy = ListNode(0)
    dummy.next = head

    var current = dummy
    while (current.next != nil) {
        if current.next!.val == val {
            current.next = current.next?.next
        } else {
            current = current.next!
        }
    }

    return dummy.next
}

File: easy/203.remove-linked-list-elements.java
public ListNode removeElements(ListNode head, int val) {
    ListNode dummy = new ListNode(0, head);

    ListNode current = dummy;
    while (current.next != null) {
        if (current.next.val == val) {
            current.next = current.next.next;
        } else {
            current = current.next;
        }
    }

    return dummy.next;
}
